lib/propolis-client/src/instance_spec/devices.rs

```diff
@@ -276,6 +313,14 @@ pub struct DeviceSpec {
     pub network_devices: BTreeMap<SpecKey, NetworkDevice>,
     pub serial_ports: BTreeMap<SpecKey, SerialPort>,
     pub pci_pci_bridges: BTreeMap<SpecKey, PciPciBridge>,
+    #[cfg(feature = "falcon")]
+    pub softnpu_pci_port: Option<SoftNpuPciPort>,
+    #[cfg(feature = "falcon")]
+    pub softnpu_ports: BTreeMap<SpecKey, SoftNpuPort>,
+    #[cfg(feature = "falcon")]
+    pub softnpu_p9: Option<SoftNpuP9>,
+    #[cfg(feature = "falcon")]
+    pub p9fs: Option<P9fs>,
```

Comment:
I'll defer to @gjcolombo on the instance spec stuff here.

Reply:
Thanks! I've left a few comments.